src/cr/cube/vector.py

```diff
@@ -239,10 +231,23 @@ def anchor(self):
         return self._base_vector.anchor
 
 
-class _BasePrunedOrHiddenVector(_BaseTransformationVector):
-    def __init__(self, base_vector, opposite_vectors):
+class PrunedVector(_BaseTransformationVector):
+    def __init__(self, base_vector, opposite_vectors, prune=True):
         self._base_vector = base_vector
         self._opposite_vectors = opposite_vectors
+        self._prune = prune
+
+    @lazyproperty
+    def _valid_elements_idxs(self):
+        """An 1D ndarray of int idxs of non-hidden values, suitable for indexing."""
+        return np.array(
+            [
+                index
+                for index, opposite_vector in enumerate(self._opposite_vectors)
+                if not opposite_vector.hidden
+            ],
+            dtype=int,
+        )
```

Comment:
Helper/implementation methods should go at the bottom of the class, in alphabetical order among the other helper methods. This is part of the important agile principle of _Seiri_ https://en.wikipedia.org/wiki/5S_(methodology)#Sort_.28Seiri.29

I fixed this in a following commit.